tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java

```diff
@@ -0,0 +1,259 @@
+    workspaceCommand.run(args);
+    String repoPath = fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os");
+    repo = gitRepoFactory.create(repoPath);
+    repo.setFakeUsersData();
+  }
+
+  private void createBlockingStub() throws IOException {
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+    server =
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor()
+            .addService(codeReviewService)
+            .build()
+            .start();
+    channel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
+    blockingStub = CodeReviewServiceGrpc.newBlockingStub(channel);
+  }
+
+  public String joinPaths(String first, String... more) {
+    return FileSystems.getDefault().getPath(first, more).toAbsolutePath().toString();
+  }
+
+  private String getWorkspaceFolder(String workspace) {
+    return joinPaths(aaBaseFolder, "ws", workspace);
+  }
+
+  private void writeFile(String contents) {
+    writeFile(TEST_FILE, contents);
+  }
+
+  private void writeFile(String filename, String contents) {
+    fileUtils.writeStringUnchecked(
+        contents, fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os", filename));
+  }
+
+  private Protos.DiffFilesResponse getExpectedResponse(String commitId) {
+    return Protos.DiffFilesResponse.newBuilder()
+        .addAllBranchInfo(
+            Collections.singleton(
+                com.google.startupos.common.repo.Protos.BranchInfo.newBuilder()
+                    .setDiffId(2)
+                    .setRepoId(REPO_ID)
+                    .addCommit(
+                        com.google.startupos.common.repo.Protos.Commit.newBuilder().setId(commitId))
+                    .build()))
+        .build();
+  }
+
+  private Protos.DiffFilesResponse getResponse(String workspace, long diffNumber) {
+    Protos.DiffFilesRequest request =
+        Protos.DiffFilesRequest.newBuilder().setWorkspace(workspace).setDiffId(diffNumber).build();
+
+    return blockingStub.getDiffFiles(request);
+  }
+
+  @Test
+  public void testFirestoreClientMock() {
+    when(firestoreClient.sum(2, 2)).thenReturn(5L);
+    assertEquals(5, codeReviewService.sum(2, 2));
+  }
+
+  @Test
+  public void testFirst() {
+    // switch workspace
+//    WorkspaceCommand workspaceCommand = component.getWorkspaceCommand();
+//    String[] args = {"workspace", TEST_WORKSPACE};
+//    workspaceCommand.run(args);
+
+    when(firestoreClient.getDocument(
+            "reviewer/data/diff/2", Protos.DiffFilesResponse.newBuilder()))
+        .thenReturn(getExpectedResponse(testFileCommitId));
+    when(firestoreClient.getDocument(
+        "/reviewer/data/last_diff_id", Protos.DiffNumberResponse.newBuilder()))
+        .thenReturn(Protos.DiffNumberResponse.newBuilder().setLastDiffId(1).build());
+
+    component.getDiffCommand().run(new String[] {"diff"});
```

Comment:
Why do you need the `DiffCommand` when we're testing `CodeReviewService,GetDiffFiles()`?

Reply:
Deleted it.